### ISSUE TITLE
Error location not returned when resolveFunction return error

### ISSUE THREAD

**Issue by thefuture2092** [graphql-go/graphql#200]
Hello,

**Actual:**
 When the resolve function of field returns an error, the error is returned by graphql.Do without the locations where the error has occured.

**Expected:** 
We should see the locations in the query where the error occured.

**How to reproduce:** 
You can use the example provided in the README and change the resolve function to return an error as follow:
```go
        fields := graphql.Fields{
		"hello": &graphql.Field{
			Type: graphql.String,
			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
				return nil, errors.New("My custom error")
			},
		},
	}
	rootQuery := graphql.ObjectConfig{Name: "RootQuery", Fields: fields}
	schemaConfig := graphql.SchemaConfig{Query: graphql.NewObject(rootQuery)}
	schema, err := graphql.NewSchema(schemaConfig)
	if err != nil {
		log.Fatalf("failed to create new schema, error: %+v", err)
	}

	// Query
	query := `
		{
			hello
		}
	`
	params := graphql.Params{Schema: schema, RequestString: query}
	r := graphql.Do(params)
	rJSON, _ := json.Marshal(r)
	fmt.Printf("%s \n", rJSON) // {“data”:{“hello”:”world”}}
```
**Current result**
```json
     {
         "data":{
           "hello":null
         },
        "errors":[
             {
                 "message":"My custom error",
                 "locations":[]
             }
        ]
     } 
```

**Expected result**
```json
    {
         "data":{
           "hello":null
         },
        "errors":[
             {
                "message":"My custom error",
                "locations":[
                   {
                      "line":3,
                      "column":4
                   }
                ]
             }
        ]
     } 
```

This is happening because in the defer part of  function resolveField in executor.go, a LocatedError is created only if the recovered value is a string. 
The fix I am thinking about is to create a NewLocatedError also when we have the recovered value is an error.

I can make a pull request if it's ok with you

**Comment by ccbrown** [comment]
I believe I fixed this as part of #363.

**Comment by chris-ramon** [comment]
Hi @thefuture2092, thanks a lot for reporting the issue, it is already fixed, confirmed by using the following working example:

```go
package main

import (
	"encoding/json"
	"errors"
	"fmt"
	"log"

	"github.com/graphql-go/graphql"
)

func main() {
	var queryObject = graphql.NewObject(graphql.ObjectConfig{
		Name: "Query",
		Fields: graphql.Fields{
			"hello": &graphql.Field{
				Type: graphql.String,
				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
					return nil, errors.New("custom error")
				},
			},
		},
	})
	schema, err := graphql.NewSchema(graphql.SchemaConfig{Query: queryObject})
	if err != nil {
		log.Fatal(err)
	}
	query := `
		{
			hello
		}
	`
	result := graphql.Do(graphql.Params{
		Schema:        schema,
		RequestString: query,
	})
	b, err := json.Marshal(result)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("%s", b)
}
```

```bash
(custom-error)-> go run main.go | jq
{
  "data": {
    "hello": null
  },
  "errors": [
    {
      "message": "custom error",
      "locations": [
        {
          "line": 3,
          "column": 4
        }
      ],
      "path": [
        "hello"
      ]
    }
  ]
}
```
